sgx-hasher/src/lib.rs

```diff
@@ -77,6 +82,64 @@ impl Hasher {
     }
 }
 
+pub fn sign(
+    hash: [u8; 32],
+    key: rsa::Rsa<Private>,
+    author: sig::Author,
+    mut contents: sig::Contents,
+) -> Result<sig::Signature> {
+    // Updates the MRENCLAVE in Signature
+    contents.update_mrenclave(hash);
+
+    // Generates signature on Signature author and contents
+    let rsa_key = PKey::from_rsa(key.clone())?;
+    let md = hash::MessageDigest::sha256();
+    let mut signer = sign::Signer::new(md, &rsa_key)?;
+    signer.update(author.as_ref())?;
+    signer.update(contents.as_ref())?;
+    let signature = signer.sign_to_vec()?;
+
+    // Generates q1, q2 values for RSA signature verification
+    let s = bn::BigNum::from_slice(&signature)?;
+    let m = key.n();
+    let mut ctx = bn::BigNumContext::new()?;
+
+    let mut q1 = bn::BigNum::new()?;
+    let mut qr = bn::BigNum::new()?;
+
+    let sig = u32::try_from(s.as_ref()).unwrap();
+    let s2 = bn::BigNum::from_u32(sig * sig)?;
+    q1.div_rem(&mut qr, &(s2), &m, &mut ctx)?;
+
+    let rem = u32::try_from(qr.as_ref()).unwrap();
+    let modl = u32::try_from(m).unwrap();
+    let q2 = (sig * rem) / modl;
+    let q2 = bn::BigNum::from_u32(q2).unwrap();
```

Comment:
We definitely should not convert to `u32` for this math. This block of code will fail at runtime because the `BigNum` will be too large to fit in `u32`.

Reply:
Good point, however `binary operation '*' cannot be applied to type 'openssl::bn::BigNum'` ... so should I go back to using the `BigNum` ops? Or is there another way I am overlooking?